controllers/upgrade_handlers.go

```diff
@@ -19,41 +19,72 @@ package controllers
 import (
 	"context"
 	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
 
 	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
 	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
+	rpmostreeclient "github.com/openshift-kni/lifecycle-agent/ibu-imager/ostreeclient"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
-func (r *ImageBasedUpgradeReconciler) handleUpgrade(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
+func updateProgressFile(progressfile, progress string) (err error) {
+	f, err := os.Create(pathOutsideChroot(progressfile))
+	if err != nil {
+		return
+	}
+	defer f.Close()
+
+	_, err = f.WriteString(progress)
+	return
+}
+
+func (r *ImageBasedUpgradeReconciler) handlePreReboot(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade, progressfile string) (err error) {
+	if err = updateProgressFile(progressfile, "started-pre-reboot"); err != nil {
```

Comment:
I think i miss something but why we are using file for progress? Everything should be in status